scripts/zones/Port_Jeuno/npcs/Sagheera.lua

```diff
@@ -332,15 +366,52 @@ entity.onTrigger = function(player, npc)
     then
         player:startEvent(313)
 
-    -- DEFAULT DIALOG (menu)
+    -- Prevent interaction until player has progressed through COP enough
+    elseif player:getCurrentMission(xi.mission.log_id.COP) < xi.mission.id.cop.GARDEN_OF_ANTIQUITY then
+        player:showText(npc, ID.text.SAGHEERA_NO_LIMBUS_ACCESS)
+
+        -- DEFAULT DIALOG (menu)
     else
         -- event parameters
+
+        local storedABCs = player:getCurrency("ancient_beastcoin")
+        local playerSagheera = player:getCharVar("SagheeraInteractions")
+        -- bitfield of menu options
+        local menu = 0
```

Comment:
Mulling over a way to simplify this logic for menu param, but works for now.  Could possible reduce this by setting the initial value of the charvar to the correct bit positions for those 3 bits, and then apply some and-logic for the rest

Reply:
That's clever. Before inverting it we could have just done a bit.band but I don't think we can anymore.

Reply:
I had assumed you were going to split the 1 time use flags off from your main mask in this pr. they don't have to be together really, and I thought that would allow you to use charvars for the one time use and get/setLocalVar on the rest

Reply:
wasn't suggesting those become localvar, was suggesting the parts that are not those, not be charvar and not be in the same mask with them

Reply:
talked to JMC in discord, I misread something and my comments can be disregarded here

Reply:
How does the latest commit look? Is that what you're thinking?

Reply:
Using a web version right now to test out ideas, is bit 2 inverted with an 11 param?  Might need to provide it 0xF